src/darwin/Framework/CHIP/templates/clusters-tests.zapt

```diff
@@ -140,6 +140,37 @@ CHIPDevice * GetPairedDevice(uint64_t deviceId)
     [self waitForExpectationsWithTimeout:kTimeoutInSeconds handler:nil];
 }
 
+
+- (void)testReuseChipClusterObject
+{
+    XCTestExpectation * expectation = [self expectationWithDescription:@"ReuseCHIPClusterObjectFirstCall"];
+
+    CHIPDevice * device = GetPairedDevice(kDeviceId);
+    dispatch_queue_t queue = dispatch_get_main_queue();
+    CHIPTestCluster * cluster = [[CHIPTestCluster alloc] initWithDevice:device endpoint:1 queue:queue];
+    XCTAssertNotNil(cluster);
+
+    [cluster test:^(NSError * err, NSDictionary * values) {
+        NSLog(@"ReuseCHIPClusterObject test Error: %@", err);
+        XCTAssertEqual(err.code, 0);
+        [expectation fulfill];
+    }];
+
+    [self waitForExpectationsWithTimeout:kTimeoutInSeconds handler:nil];
+
+    expectation = [self expectationWithDescription:@"ReuseCHIPClusterObjectSecondCall"];
+
+    // Reuse the CHIPCluster Object for multiple times.
```

Comment:
It would be a good idea to reuse a number of times larger than `CHIP_MAX_NUM_COMMAND_SENDER` so that we catch cases when we're leaking senders.  Followup is OK for this.

Reply:
#7074 Filed.